### ISSUE TITLE
Unraid Docker Template

### ISSUE THREAD

**Issue by git-Contributor** [izderadicka/audioserve#65]
Hi, 
I wanted to use audioserve on Unraid and as I found in [issue](https://github.com/izderadicka/audioserve/issues/38) and also on [reddit](https://libredd.it/r/unRAID/comments/j7ccmj/help_creating_a_template_for_audioserve/) I am not alone. Since I was struggling as others, I wanted to share working template so you can include it in repo and help spread this amazing app :)

(Addes as zip since XML is not allowed)
[audioserve_unraid.zip](https://github.com/izderadicka/audioserve/files/7274018/audioserve_unraid.zip)

There is also possibile to add SSL using AUDIOSERVE_SLL_KEY and AUDIOSERVE_SLL_KEY_PASSWORD variables, but I had issue with Lets Encrypt Certificate on official android app, so I am not using SSL currently.

**Comment by izderadicka** [comment]
Thanks I'll include in documentation. 
Concerning Let's Encrypt  certificate - it's interesting as I'm using it with android without any problem. You can check demo site https://audioserve.zderadicka.eu/   which is also using Let's Encrypt (just try to connect there from android app).

I do not know much about Unraid, but just watching the XML I see there just one volume mapping. I'd highly recommend to include there another volume, which will contain audioserve writable data like shard secret, position db,  transcoding cache, collections cache (to be there in next version) -  in docker it's mapped like this `-v /path/for/audioserve-data:/home/audioserve/.audioserve` - I think with your configuration it's in transient container file system, so a) it is slow (this will be especially important for future ), b) you loose the data with each container restart.

**Comment by git-Contributor** [comment]
Yeah, I was not sure if /home/audioserve/.audioserve is needed. I will add it in the template, it will just map this path to /mnt/user/appdata/audioserve. I can also add the SSL, if we can figure it out:

- When I log in from browser, it is OK, I have lock next to the address and everything seems to be ok (also it is same certificate I am using for jellyfin and there I have no issue):
![image](https://user-images.githubusercontent.com/91840689/135769678-b5fc0160-3f6a-4656-aeab-25c21d73cc7e.png)
- When I try to log in from android (via latest app) I got error in docker log:
[2021-10-03T19:58:30Z ERROR audioserve::tls] Error when accepting TLS connection error:14094416:SSL routines:ssl3_read_bytes:sslv3 alert certificate unknown:ssl/record/rec_layer_s3.c:1543:SSL alert number 46

Any idea why is that happening?

**Comment by izderadicka** [comment]
From error I understand that you are using TLS on audioserve right?

What version is your android? Older versions of openssl have issues with
Let's Encrypt see
https://www.openssl.org/blog/blog/2021/09/13/LetsEncryptRootCertExpire/

This error
unknown:ssl/record/rec_layer_s3.c:1543:SSL alert number 46
means that certificate is rejected in client, android in your case. If
serving TLS from audioserve assure that full certificate path is provided
in supplied .p12 file. But this would cause problem also in browser unless
you have exception there.

You can also check android log via USB to see why exactly certificate is
rejected.

I.

**Comment by izderadicka** [comment]
@git-Contributor Do you have updated version of unraid config file?

**Comment by git-Contributor** [comment]
Attached: [audioserve_unraid.zip](https://github.com/izderadicka/audioserve/files/7364351/audioserve_unraid.zip)
I just added mounted /home/audioserve/.audioserve to /mnt/user/appdata/audioserve, which is default folder for docker config/data path. The SSL was not working for me so it is not included in the template, but if anybody will need it, they can easily add it there using your documentation.
